### ISSUE TITLE
fault tolerant training with dataloader

### ISSUE THREAD

**Issue by yuvalkirstain** [pytorch/pytorch#97622]
### 🚀 The feature, motivation and pitch

When training with a dataloader, we might stop training in the middle of the run, and start it again later. Then, we usually want to start when we finished last time. Currently, I need to go over all of the data I went over in the previous run, which can be very slow. It would be nice to change it.

### Alternatives

If we could save the state of the dataloader, and then load it, or if we could have a skipping mode that returns dummy batches until we get to the step we stopped in the previous run, it will probably lead to much faster results, when continuing a run that was stopped.

**Comment by pat749** [comment]
use PyTorch's built-in functionality to save and load checkpoint files. Checkpoint files contain the state of the model, optimizer, and scheduler, as well as the current epoch and iteration numbers. You can also save and load the state of the dataloader by adding it to the checkpoint file.

To do this, you can create a dictionary containing the state of the model, optimizer, scheduler, and dataloader, and save it using **torch.save**. When resuming training, you can load the checkpoint file using **torch.load**, which will restore the state of the dataloader to where it was when the training was stopped.
